pkg/trace/writer/sender.go

```diff
@@ -247,6 +247,10 @@ func (s *sender) sendPayload(p *payload) {
 			return
 		}
 		atomic.AddInt32(&s.attempt, 1)
+
+		if r := atomic.AddInt32(&p.retry, 1); shouldWarnRetry(r) {
+			log.Warnf("Retried payload %d times, got error: %s", r, err.Error())
```

Comment:
```suggestion
			log.Warnf("Retried payload %d times: %s", r, err.Error())
```
Maybe this is nicer?